cleanrl/muesli_atari_envpool_async_jax_scan_impalanet_machado.py

```diff
@@ -0,0 +1,1393 @@
+        action = jnp.argmax(logits - jnp.log(-jnp.log(u)), axis=1)
+        logprob = jax.nn.log_softmax(logits)[jnp.arange(action.shape[0]), action]
+        return action, logprob, lstm_carry, key
+
+    @jax.jit
+    def normalize_logits(logits: jnp.ndarray) -> jnp.ndarray:
+        """Normalize thhe logits.
+
+        # https://gregorygundersen.com/blog/2020/02/09/log-sum-exp/
+        """
+        logits = logits - jax.scipy.special.logsumexp(logits, axis=-1, keepdims=True)
+        logits = logits.clip(min=jnp.finfo(logits.dtype).min)
+        return logits
+
+    @jax.jit
+    def reward_transform(r: jnp.ndarray) -> jnp.ndarray:
+        """Transform the reward.
+
+        h(x): R -> R, epsilon > 0
+        x -> sgn(x)(sqrt(abs(x)+1)-1) + epsilon * x
+
+        The function is applied element-wise.
+
+        See the last paragraph of Hessel et al., 2021. Muesli paper. Section 4.5. for details.
+        Alternatively, see Pohlen et al., 2018. Observe and Look Further: Achieving Consistent Performance on Atari.
+        - Section 3.2: how to use it
+        - Proposition A.2: the inverse and some properties of the transformation.
+        """
+        eps = args.reward_transformation_epsilon
+        return jnp.sign(r) * (jnp.sqrt(jnp.abs(r) + 1) - 1) + eps * r
+
+    @jax.jit
+    def inverse_reward_transform(tr: jnp.ndarray) -> jnp.ndarray:
+        """De-transform the reward.
+
+        h^-1(x): R -> R, epsilon > 0
+        x -> sgn(x)(((sqrt(1 + 4*epsilon*(abs(x)+1+epsilon)) - 1)/(2*epsilon))^2-1)
+
+        The function is applied element-wise.
+
+        See the last paragraph of Hessel et al., 2021. Muesli paper. Section 4.5. for details.
+        Alternatively, see Pohlen et al., 2018. Observe and Look Further: Achieving Consistent Performance on Atari.
+        - Section 3.2: how to use it
+        - Proposition A.2: the inverse and some properties of the transformation.
+        """
+        eps = args.reward_transformation_epsilon
+        num = jnp.sqrt(1 + 4 * eps * (jnp.abs(tr) + 1 + eps)) - 1
+        denom = 2 * eps
+        return jnp.sign(tr)((num / denom) ** 2 - 1)
+
+    @jax.jit
+    def unroll_model(carry, act):
+        carry, pred_latent_state = dynamics.apply(agent_state.params.dynamics_params, carry, act)
+        pred_reward, pred_value = reward_value_model.apply(agent_state.params.reward_value_model_params, pred_latent_state)
+        pred_policy = normalize_logits(policy_model.apply(agent_state.params.policy_model_params, pred_latent_state))
+        return carry, (pred_reward, pred_value, pred_policy)
+
+    @jax.jit
+    def compute_pred_q_and_policy(carry, act):
+        _, pred_latent_state = dynamics.apply(agent_state.params.dynamics_params, carry, act)
+        pred_reward, pred_value = reward_value_model.apply(agent_state.params.reward_value_model_params, pred_latent_state)
+        return carry, (pred_reward, pred_value)
+
+    @jax.jit
+    def reanalyze_prior_policies_values_latent_states(carry, x):
```

Comment:
if they are being used in a `jax.scan`, `jax.jit` operator should not be necessary.

Reply:
Fixed.